src/tools/chdman.cpp

```diff
@@ -1016,6 +1016,72 @@ static void guess_chs(std::string *filename, uint64_t filesize, int sectorsize,
 }
 
 
+//-------------------------------------------------
+//  reopen_chd_with_parent - find a parent chd in the
+//  directory specified
+//-------------------------------------------------
+
+static chd_error reopen_chd_with_parents(chd_file &input, const char* chd_path, const std::string parents_dirpath, bool writeable = false)
+{
+	if (!input.opened())
+  {
+		input.open(chd_path, writeable);
+	}
+	chd_file *parent = new chd_file;
+	chd_file *child = NULL;
+
+	int depth = 0;
+	std::string paths[8];
+	std::string filepath;
+	util::sha1_t parent_sha1 = input.parent_sha1();
+	chd_error err;
+	while (true)
+	{
+		osd::directory::ptr dir = osd::directory::open(parents_dirpath);
+		const osd::directory::entry *entry;
+		while((entry = dir->read()) != nullptr)
+		{
+			filepath = parents_dirpath + PATH_SEPARATOR + entry->name;
```

Comment:
You should be checking that it’s actually a file and that it has the `.chd` filename extension here.

Reply:
File check, OK. About the extension, does osdcore/mame handle a case-insensitive globbing? Or will we have to implement it ourselves? It seems like a trivial thing to implement, but dealing with C/C++, UTF-8, and case-sensitivity, is like a defusing a time-bomb. Well, at least it was when I used to use C++ extensively. I googled quickly, but still couldn't find a sane way of dealing with it with std.

If osdcore/mame doesn't handle it, can we simplify and check only for `.chd` and `.CHD`, and not deal with case-conversion at all?